### PR TITLE
Heba salem react.js week1 homework

### PR DESCRIPTION
**Pull request by HebaMak**
eCommerce app link on Netlify:

https://incomparable-beignet-a0fda3.netlify.app/

### CODE REVIEW

**Comment by HebaMak** [comment]
eCommerce app link on Netlify;

https://incomparable-beignet-a0fda3.netlify.app/

**Comment by HebaMak** [comment]
eCommerce app link on Nitlify:

https://incomparable-beignet-a0fda3.netlify.app/

File: week1/project/ecommerce/src/components/Products.jsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  const {products} = useContext(provideContext)
+
+  return (
+    <>
+      <h1 className="title">Products</h1>
+      <Categories />
+      <div className='products_container'>
+        {
+          products.map((product, index) => (
+          <Product key={index} product={product}/>
```
Comment (hadeel43) [inline week1/project/ecommerce/src/components/Products.jsx:19]: Looks good, but I prefer to not use index as a key of array - it has negative performance issues, instead we should try to find some unique and permanent value for keys

https://robinpokorny.medium.com/index-as-a-key-is-an-anti-pattern-e0349aece318

File: week1/project/ecommerce/src/components/Categories.jsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import allCategories from "../fake-data/all-categories"
+
+import Category from "./Category"
+
+function Categories() {
+  
+  return (
+    <div className="categories_container"> 
+      {
+        allCategories.map((category, index) => (
```
Comment (hadeel43) [inline week1/project/ecommerce/src/components/Categories.jsx:10]: Same for the use of index :)

File: week1/project/ecommerce/src/components/Products.jsx (hunk 1 above)
Comment (HebaMak) [reply week1/project/ecommerce/src/components/Products.jsx:19]: done

File: week1/project/ecommerce/src/components/Categories.jsx (hunk 2 above)
Comment (HebaMak) [reply week1/project/ecommerce/src/components/Categories.jsx:10]: categories array, has no ids and we are allowed to edit it,
once we work with the API I will change it to the category id

**Pull request closed without merging**